buildSrc/src/test/java/io/micronaut/guides/core/SourceMacroSubstitutionTest.java

```diff
@@ -18,7 +17,7 @@ class SourceMacroSubstitutionTest {
 
     @Test
     void testSubstitute(){
-        String str = "source:GithubConfiguration[]";
+        String str = "source:GithubConfiguration[]\n";
```

Comment:
why do you need to add a \n line character. 

```suggestion
        String str = "source:GithubConfiguration[]\n";
```

Reply:
this looks like a bad smell to me.

Reply:
If we have multi-line inputs, the different lines will be separated by a \n, so it was to test that. If you look at the other tests you will see that input without \n is also tested and works.